utilities/backupable/backupable_db.cc

```diff
@@ -281,14 +298,20 @@ class BackupEngineImpl : public BackupEngine {
     return GetSharedChecksumDirRel() + "/" + (tmp ? "." : "") + file +
            (tmp ? ".tmp" : "");
   }
-  inline std::string GetSharedFileWithChecksum(const std::string& file,
-                                               const uint32_t checksum_value,
-                                               const uint64_t file_size) const {
+  // If UseNewNaming() && !db_session_id.empty(), backup SST filenames consist
+  // of file_number, crc32c, db_session_id.
+  // Otherwise, backup SST filenames consist of file_number, crc32c, file_size
+  inline std::string GetSharedFileWithChecksum(
+      const std::string& file, const uint32_t checksum_value,
+      const uint64_t file_size, const std::string& db_session_id) const {
     assert(file.size() == 0 || file[0] != '/');
     std::string file_copy = file;
-    return file_copy.insert(file_copy.find_last_of('.'),
-                            "_" + ROCKSDB_NAMESPACE::ToString(checksum_value) +
-                                "_" + ROCKSDB_NAMESPACE::ToString(file_size));
+    const std::string suffix = UseNewNaming() && !db_session_id.empty()
+                                   ? db_session_id
```

Comment:
I'm still concerned we aren't fully guaranteeing that the contents of db_session_id is legal/appropriate for a file name. ('/' illegal, ' ' not appropriate)  Perhaps this is just a matter of saying in the contract of DB::GetDbSessionId that the characters are usable in a file name. That way, someone writing an alternate implementation will be aware of the expectation.

And we'll deal with this problem again with user checksums, which can be arbitrary byte data in a string.

Reply:
Good point! I think we could also add some kind of sanitizer for the backup file name later. (I think the session id and db id generated by RocksDB consist of only hexadecimal digits and "-")